python/pyspark/java_gateway.py

```diff
@@ -97,8 +99,18 @@ def preexec_func():
                 popen_kwargs['preexec_fn'] = preexec_func
                 proc = Popen(command, **popen_kwargs)
             else:
+                # If an argument contains a character that needs escaping (e.g. an 
+                # ampersand) we need to apply a double quoting mechanism to correctly 
+                # pass the value through the submit shell scripts.
+                def quote(s):
+                    r, count = subn(r"([&<>()|,;^])", r"^^^\1", s)
+                    if count:
+                        return '"""' + r + '"""'
+                    return s
```

Comment:
Can we put this in a utils function and or is there a library that does this escaping? I imagine there's a bunch of random values and I'd rather us get them all.

Reply:
I have looked and not found any ...